pkg/controller/namespace/namespace_controller.go

```diff
@@ -332,3 +336,17 @@ func deletePersistentVolumeClaims(kubeClient client.Interface, ns string) error
 	}
 	return nil
 }
+
+func deleteLocks(kubeClient client.Interface, ns string) error {
```

Comment:
Should check with decarr where is that recursive delete @... 
This is the same iterative problem.